readme.md

```diff
@@ -343,15 +343,60 @@ for ($i = 0; $i < 10; $i++) {
 print_r($values); // [1, 4, null, 9, 5, null, null, 4, 6, null]
 
 // optional() accepts a weight argument to specify the probability of receiving the default value.
-// 0 will always return the default value; 1 will always return the provider. Default weight is 0.5 (50% chance).
-$faker->optional($weight = 0.1)->randomDigit; // 90% chance of NULL
-$faker->optional($weight = 0.9)->randomDigit; // 10% chance of NULL
+// --------------------------------------------------------
+// 2 value sets are possible
+// --------------------------------------------------------
+// First set is between 0 and 1 (Fractional representation)
+// --------------------------------------------------------
+// 0 will always return the default value
+// Anything else BELOW 1 will be converted to a percentage
+// 
+// --------------------------------------------------------
+// Second set is between 1 and 100 (Percentage representation)
+// --------------------------------------------------------
+// Value MUST be an integer
+// 0 will always return the default value
+// 1 will return the provider's value 1% of the time
+// 100 will return the provider's value 100% of the time
+// --------------------------------------------------------
+// Default weight is 0.5 (50% chance).
+
+// First set examples
+$faker->optional($weight = 0.05)->randomDigit; // 95% chance of NULL, 5% chance of provider
+$faker->optional($weight = 0.1)->randomDigit; // 90% chance of NULL, 10% chance of provider
+$faker->optional($weight = 0.9)->randomDigit; // 10% chance of NULL, 90% chance of provider
+
+// Second set examples
+$faker->optional($weight = 1)->randomDigit; // 1% chance of NULL, 99% chance of provider
+$faker->optional($weight = 10)->randomDigit; // 90% chance of NULL, 10% chance of provider
+$faker->optional($weight = 100)->randomDigit; // 0% chance of NULL, 100% chance of provider
+
+// PITFALLS
+// --------------------------------------------------------
+// Get a half a percent (0.5%) chance
+$faker->optional($weight = 0.5)->randomDigit; // This is not 0.5%, this is 50% because it falls within first set
+$faker->optional($weight = 0.005)->randomDigit; // This is a 99.5% chance of NULL or 0.5% chance of provider
+// --------------------------------------------------------
+// Get 100% chance using a true/false or 0/1 approach
+$weight = $faker->randomElement([0, 1]);
+$faker->optional($weight)->randomDigit; // This will yield randomDigit 0% or 1% of chances, not 0% or 100%
+$faker->optional($weight * 100)->randomDigit; // This will yield randomDigit 0% or 100% of chances
```

Comment:
Perhaps we should add a note that this behaviour will be removed in `2.0`?

See https://github.com/fzaninotto/Faker/blob/3f74f49dd7c53af959b3708a64d9232db271f985/src/Faker/Provider/Base.php#L547-L551.